src/Orleans.Streaming/PersistentStreams/QueueStreamDataStructures.cs

```diff
@@ -40,18 +41,24 @@ public StreamConsumerData(GuidId subscriptionId, InternalStreamId streamId, IStr
 
         internal void SafeDisposeCursor(ILogger logger)
         {
-            try
+            if (Cursor is { } cursor)
             {
-                if (Cursor != null)
+                Cursor = null;
+                // kill cursor activity and ensure it does not start again on this consumer data.
+                try
                 {
-                    // kill cursor activity and ensure it does not start again on this consumer data.
-                    Utils.SafeExecute(Cursor.Dispose, logger,
-                        () => String.Format("Cursor.Dispose on stream {0}, StreamConsumer {1} has thrown exception.", StreamId, StreamConsumer));
+                    cursor.Dispose();
+                }
+                catch (Exception ex)
```

Comment:
I think we're sometimes too defensive, although I suppose the method is called *Safe*DisposeCursor...